service/module.go

```diff
@@ -27,8 +27,7 @@ import (
 
 // ModuleProvider provides Modules.
 type ModuleProvider interface {
-	// DefaultName is the name that will be used for a new Module
-	// if no name is given as a ModuleOption.
+	// DefaultName returns the module name
```

Comment:
Either comment is incorrect, or method name should change.
DefaultName that returns module name sounds weird..